### PR TITLE
Return StackName of managed Nodegroups created through CF

### LINKED ISSUE #3606: StackName comes up blank
When running the following command, eksctl get nodegroup --cluster -o json, "StackName" comes up blank. It works when I change eksctl from 0.45.0 to 0.44.0
@brentley

### PR DESCRIPTION
**Pull request by nikimanoledaki**
### Description

Closes #3606 

There was some slight regression when the code was refactored to make calls about Nodegroups directly to the ASG and less calls to Cloud Formation in PR #3543. 

The bug this PR fixes: `eksctl get nodegroup --cluster <cluster-name> -o json` was no longer returning the name of managed Nodegroups. 

Note that this bug did not apply to unmanaged Nodegroups, where the StackName was being returned.

Also note that managed Nodegroups created through the EKS console do not make use of Cloud Formation, so there would be no StackName to return for Nodegroups created that way anyway.

**Demo**
In the output below, the 2 Nodegroups (managed and unmanaged) that were created through eksctl return a StackName. The Nodegroup that was created through the console returns `""` as a `StackName` since it was not created through a CF stack.
```
./eksctl get nodegroup --cluster scrumptious-hideout-1620648061 -o json
[
    {
        "StackName": "eksctl-scrumptious-hideout-1620648061-nodegroup-ng-3267166a",
        "Cluster": "scrumptious-hideout-1620648061",
        "Name": "ng-3267166a",
        "Status": "CREATE_COMPLETE",
        "MaxSize": 2,
        "MinSize": 2,
        "DesiredCapacity": 2,
        "InstanceType": "m5.large",
        "ImageID": "ami-01a9605bd94e3d03d",
        "CreationTime": "2021-05-10T12:15:22.632Z",
        "NodeInstanceRoleARN": "arn:aws:iam::749339757188:role/eksctl-scrumptious-hideout-162064-NodeInstanceRole-14CV67VRHUQ19",
        "AutoScalingGroupName": "eksctl-scrumptious-hideout-1620648061-nodegroup-ng-3267166a-NodeGroup-2PT2TL7COILL"
    },
    {
        "StackName": "",
        "Cluster": "scrumptious-hideout-1620648061",
        "Name": "exciting-ng",
        "Status": "ACTIVE",
        "MaxSize": 2,
        "MinSize": 2,
        "DesiredCapacity": 2,
        "InstanceType": "t3.medium",
        "ImageID": "AL2_x86_64",
        "CreationTime": "2021-05-10T12:51:39.054Z",
        "NodeInstanceRoleARN": "arn:aws:iam::749339757188:role/eksctl-scrumptious-hideout-162064-NodeInstanceRole-14CV67VRHUQ19",
        "AutoScalingGroupName": "eks-18bcab2a-9e8f-747b-fbb2-0a6b0b19bae4"
    },
    {
        "StackName": "eksctl-scrumptious-hideout-1620648061-nodegroup-ng-982df192",
        "Cluster": "scrumptious-hideout-1620648061",
        "Name": "ng-982df192",
        "Status": "ACTIVE",
        "MaxSize": 2,
        "MinSize": 2,
        "DesiredCapacity": 2,
        "InstanceType": "m5.large",
        "ImageID": "AL2_x86_64",
        "CreationTime": "2021-05-10T12:23:44.787Z",
        "NodeInstanceRoleARN": "arn:aws:iam::749339757188:role/eksctl-scrumptious-hideout-162064-NodeInstanceRole-ACO3XG6R64SB",
        "AutoScalingGroupName": "eks-5abcab1d-d719-c6da-f691-78ca2e437d86"
    }
]%   
```

### Checklist
- [x] Added tests that cover your change (if possible)
- [x] Added/modified documentation as required (such as the `README.md`, or the `userdocs` directory)
- [x] Manually tested
- [x] Made sure the title of the PR is a good description that can go into the release notes
- [x] (Core team) Added labels for change area (e.g. `area/nodegroup`) and kind (e.g. `kind/improvement`)

### BONUS POINTS checklist: complete for good vibes and maybe prizes?! :exploding_head:
- [x] Backfilled missing tests for code in same general area :tada:
- [x] Refactored something and made the world a better place :star2:

### CODE REVIEW

**Comment by aclevername** [comment]
:clap: amazing PR description :clap:

File: pkg/actions/nodegroup/get.go [hunk 1]
```diff
@@ -33,6 +34,12 @@ func (m *Manager) GetAll() ([]*manager.NodeGroupSummary, error) {
 			return nil, err
 		}
 
+		var stack *cloudformation.Stack
+		stack, err = m.stackManager.DescribeNodeGroupStack(*managedNodeGroup)
+		if err != nil {
+			stack = &cloudformation.Stack{}
+		}
```
[the remark this answers is not in the document]
Comment (nikimanoledaki) [reply pkg/actions/nodegroup/get.go:41]: ah ok! I tested this manually quite a bit yesterday with the nodegroup created via the console, to check the cases where it returned a nil pointer dereference, until it didn't! unfortunately we can't do integration tests with resources created in the console can we?

[the remark this answers is not in the document]
Comment (aclevername) [reply pkg/actions/nodegroup/get.go:41]: > ah ok! I tested this manually quite a bit yesterday with the nodegroup created via the console, to check the cases where it returned a nil pointer dereference, until it didn't! unfortunately we can't do integration tests with resources created in the console can we?

we can! but its not as simple. We have one test suite that currently does this: https://github.com/weaveworks/eksctl/blob/0043ae974591fb0085ae4ccb04d344745eeb1bc9/integration/tests/unowned_cluster/unowned_cluster_test.go#L460-L470

It takes to the API directly (essentially same as console) to create non-eksctl created managed stacks

**Review by aclevername — APPROVED** [review]
LGTM!

**Pull request merged**